src/array/cpu/spmat_op_impl_coo.cc

```diff
@@ -433,38 +438,38 @@ CSRMatrix UnSortedSparseCOOToCSR(const COOMatrix &coo) {
   IdType *const Sx = static_cast<IdType *>(sorted_data->data);
   IdType *const Si = static_cast<IdType *>(sorted_data_pos->data);
 
+  // Lower number of threads if cost of parallelization is grater than gain
+  // from making calculation parallel.
+  const int64_t min_chunk_size = 1000;
+  const int64_t num_threads_for_batch = 2 + (NNZ + N) / min_chunk_size;
```

Comment:
Can you add in comments why we want the minimum number of threads used to be 2 instead of 1?

Reply:
That makes sense. I'd be in favor of having the minimum be 1 even though it should never be hit with the current code, so that if refactoring happens in the future this function behaves as expected.

I'll mark this as resolved as I think it is acceptable as is if you prefer.